File: Day_5/Two-Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> vp;
        for(int i = 0; i < nums.size(); i++){
            vp.push_back({nums[i],i});
        }
        sort(vp.begin(),vp.end());
        
        for(int i = 0; i < vp.size(); i++){
            int req = target - vp[i].first;
            
            // Binary search for the required value
            int left = 0;
            int right = vp.size() - 1;
            
            while(left <= right){
                int mid = left + (right - left) / 2;
                
                if(vp[mid].first == req && mid != i){
                    // Found the required value
                    return {vp[i].second, vp[mid].second};
                }
                else if(vp[mid].first < req){
                    left = mid + 1;
                }
                else{
                    right = mid - 1;
                }
            }
        }
        
        // If no solution is found (though the problem guarantees one exists)
        return {};
    }
};

File: Day_111/Two-Sum.py
1class Solution:
2    def twoSum(self, nums: List[int], target: int) -> List[int]:
3        ind_dict = {}
4        
5        for ind in range(len(nums)):
6            required_value = target - nums[ind]
7
8            if required_value in ind_dict and ind_dict[required_value] != ind:
9                return [ind_dict[required_value], ind]
10            
11            ind_dict[nums[ind]] = ind
12